core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -218,12 +223,16 @@ private int positionedReadInternal(long pos, byte[] b, int off, int len) throws
         break;
       }
       long blockId = mStatus.getBlockIds().get(Math.toIntExact(pos / mBlockSize));
-      BlockInStream stream = null;
       try {
-        stream = mBlockStore.getInStream(blockId, mOptions, mFailedWorkers);
+        if (mCachedPositionedReadStream == null) {
```

Comment:
can you add a comment as to why use a separate instream?

Reply:
sgtm